crates/router/src/core/webhooks/outgoing.rs

```diff
@@ -457,13 +473,50 @@ fn raise_webhooks_analytics_event(
         .and_then(api::OutgoingWebhookContent::get_outgoing_webhook_event_content)
         .or_else(|| get_outgoing_webhook_event_content_from_event_metadata(event.metadata));
 
+    // Fetch updated_event from db
+    let updated_event = state
+        .store
+        .find_event_by_merchant_id_event_id(
+            key_manager_state,
+            &merchant_id,
+            &event_id,
+            merchant_key_store,
+        )
+        .await
+        .attach_printable_lazy(|| format!("event not found for id: {}", &event_id))
+        .map_err(|e| {
+            logger::error!("{:?}", e);
+            e
+        })
+        .ok();
+
+    // Get status_code from webhook response
+    let status_code = if let Some(updated_event) = updated_event {
```

Comment:
Since we've used `and_then()` elsewhere in this function, we can use it here as well? Something like:
`let status_code = updated_event.and_then(|...| ...);`